Sources/AccountSDKIOSWeb/User/User.swift

```diff
@@ -153,15 +153,28 @@ extension User {
     func refreshTokens(completion: @escaping (Result<UserTokens, RefreshTokenError>) -> Void) {
         self.refreshHandler.refreshWithoutRetry(user: self, completion: completion)
     }
+    
+    func retrieveTokens() {
+        if let newTokens = client.retrieveTokens() {
```

Comment:
Would it make sense to only read the tokens from the keychain, if `self.tokens` somehow has ended up being nil at this point? 
Otherwise it looks like the tokens will be read from keychainStorage for every single request now, which seems like wasted time in terms of performance.

Reply:
Can this method be run from multiple threads? If so `self.tokens` needs to by synced

Reply:
Also this will always override `self.tokens` even if they are non nil